### PR TITLE
Create Lustre integration test

### PR DESCRIPTION
**Pull request by rohitramu**

### CODE REVIEW

**Review by tpdownes — CHANGES_REQUESTED** [review]
Please address comments.

File: tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@
 - name: Test partition mounts, multi-node creation, & placement (when on)
   register: srun_mounts
   changed_when: srun_mounts.rc == 0
-  ansible.builtin.command: "srun -N 2 -p {{ item }} mount"
+  ansible.builtin.command: "srun -N {{ custom_vars.num_slurm_nodes | default(2) }} -p {{ item }} mount"
```
[the remark this answers is not in the document]
Comment (rohitramu) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml:58]: That's correct.  I don't want to change existing tests, but at the same time I want to provide an override for this (previously hardcoded) value.

Is there a better way to do this?

File: tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  register: write_file_output
+
+- name: Print console output
+  ansible.builtin.debug:
+    msg: "{{ write_file_output }}"
+
+- name: Ensure a file exists for each OS type
+  with_items: "{{ custom_vars.partitions }}"
+  ansible.builtin.shell: |
+    test $(cat {{ custom_vars.output_dir }}/{{ item }}.txt) == '{{ item }}' && echo 'Success!' || exit 1
```
[the remark this answers is not in the document]
Comment (rohitramu) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml:34]: That makes sense.  I've now removed the on-success behavior.

File: tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  with_items: "{{ custom_vars.partitions }}"
+  ansible.builtin.shell: |
+    srun -N 1 -p '{{ item }}' bash -c 'echo {{ item }} > {{ custom_vars.output_dir }}/{{ item }}.txt'
+  register: write_file_output
+
+- name: Print console output
+  ansible.builtin.debug:
+    msg: "{{ write_file_output }}"
+
+- name: Ensure a file exists for each OS type
```
[the remark this answers is not in the document]
Comment (rohitramu) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml:31]: Oh that's right, thanks.

File: tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: Create output folder
+  ansible.builtin.shell: |
+    [[ ! -d {{ custom_vars.output_dir }} ]] && sudo mkdir -m a+w {{ custom_vars.output_dir }}
```
[the remark this answers is not in the document]
Comment (rohitramu) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml:19]: Oh interesting, thanks for the explanation!  I've made both changes.

File: tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-vm.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+- name: Ensure a file exists for each OS type
+  with_items: "{{ custom_vars.vm_os_types }}"
+  ansible.builtin.shell: |
+    test $(cat {{ custom_vars.output_dir }}/{{ item }}.txt) == '{{ item }}' && echo 'Success!' || exit 1
```
[the remark this answers is not in the document]
Comment (rohitramu) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-vm.yml:19]: Fixed.

File: tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml [hunk 6]
```diff
@@ -16,19 +16,24 @@
...
 - name: Print console output
   ansible.builtin.debug:
     msg: "{{ write_file_output }}"
 
-- name: Ensure a file exists for each OS type
+- name: Ensure a file exists for each partition
   with_items: "{{ custom_vars.partitions }}"
   ansible.builtin.shell: |
-    test $(cat {{ custom_vars.output_dir }}/{{ item }}.txt) == '{{ item }}' && echo 'Success!' || exit 1
+    test $(cat {{ custom_vars.output_dir }}/{{ item }}.txt) == '{{ item }}' || exit 1
```
Comment (tpdownes) [inline tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml:37]: Sorry to nitpick here, but `test "a" == "b"` already exits 1. Unless this is doing something valuable that I'm not catching onto, please remove it.

File: tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tpdownes) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml:58]: I'll suggest something offline but it should not block this PR.

**Review by tpdownes — APPROVED** [review]
I suggest you make the one last change I'm requesting (about "exit 1") and then merge.

File: tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml (hunk 6 above)
Comment (rohitramu) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml:37]: Oh I see, sorry I misunderstood the first time.

Comment (rohitramu) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-lustre-slurm.yml]: Fixed.

**Pull request merged**